### PR TITLE
Migrate HqDeploy to SQL (1 of 3)

### PR DESCRIPTION
**Pull request by orangejenny**
Follows [steps in docs](https://github.com/dimagi/commcare-hq/blob/master/docs/couch_to_sql_models.rst#practical-steps). Skips the domain deletion step since this isn't a domain-specific model.

### CODE REVIEW

File: corehq/apps/hqadmin/migrations/0010_sqlhqdeploy.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('hqadmin', '0009_auto_20170315_1322'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='SQLHqDeploy',
```
Comment (snopoke) [inline corehq/apps/hqadmin/migrations/0010_sqlhqdeploy.py]: how about changing the SQL table name to just `hqadmin_deploy` (or something without the `SQL`)

Comment (orangejenny) [reply corehq/apps/hqadmin/migrations/0010_sqlhqdeploy.py]: Updated the table name in 0c02021d0b18dca6735d9a69a2a887402b56aa77

I'm not that keen on the class name starting with "SQL" either, might take a look at renaming it back to `HqDeploy` in a later PR.

**Review by dannyroberts — APPROVED** [review]

**Review by kaapstorm — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Review by esoergel — APPROVED** [review]

File: corehq/apps/hqadmin/migrations/0010_sqlhqdeploy.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/hqadmin/migrations/0010_sqlhqdeploy.py]: @snopoke I've been having a devil of a time getting the table name change deployed to staging. Deploy throws `django.db.utils.ProgrammingError: relation "hqadmin_hqdeploy" does not exist`, ultimately coming from [this line](https://github.com/dimagi/commcare-hq/blob/master/corehq/sql_db/__init__.py#L126) in `check_db_tables`.

I put a debug statement in `check_db_tables` and then ran `showmigrations`, and it does look like the check is running before the migration that creates the table has run. I've tried removing the `db_table` so the model uses the default table but it doesn't make a difference; I can't seem to deploy any new SQL model to staging now. But the `sql_db` code hasn't changed that recently and I've definitely deployed similar PRs in the past few weeks. Any idea what might be going on?

Comment (snopoke) [reply corehq/apps/hqadmin/migrations/0010_sqlhqdeploy.py]: That's odd. This command should only be called right at the end of deploy: https://github.com/dimagi/commcare-cloud/blob/76dd0814990400cf8cb5d35179bf9f0ff260e978/src/commcare_cloud/fab/fabfile.py#L526

Where in the deploy process was it failing?

File: corehq/apps/hqadmin/models.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ class SQLHqDeploy(models.Model):
     diff_url = models.CharField(max_length=126, null=True)
     couch_id = models.CharField(max_length=126, null=True, db_index=True)
 
-    class Meta:
+    class Meta(object):
```
Comment (millerdev) [inline corehq/apps/hqadmin/models.py]: Curious why this was required? I thought it wasn't in Python 3

**Review by millerdev — APPROVED** [review]

File: corehq/apps/hqadmin/models.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/hqadmin/models.py]: It isn't required, just an incidental change made while troubleshooting.

File: corehq/apps/hqadmin/migrations/0010_sqlhqdeploy.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/hqadmin/migrations/0010_sqlhqdeploy.py]: Fixed! It was indeed failing at the end, during perform_system_checks, even though the migration hadn't errored out. I must have gotten staging into a bad state - it had a record of the migrations being run even but the table was not longer there, so deleting the migrations from `django_migrations` and then re-running them fixed it.

**Review by millerdev — APPROVED** [review]

**Pull request merged**